### PR TITLE
ENDOC-710-tomcat-default edits

### PR DESCRIPTION
**Pull request by jyunmitch**
Need more assistance/information for:
vuepress/docs/next/docs/reference/deployment-structure.md
vuepress/docs/next/tutorials/devops/build-core-image.md
vuepress/docs/next/tutorials/devops/change-default-datasource.md

The last item, Change Default Datasources and Connections Tutorial in particular, needs a serious overhaul. @pietrangelo, is the tutorial still necessary and does the process seem to flow properly? Can you review and give us your suggestions? There are also some more specific questions in the tutorial that needs checking or new addresses/names.

### CODE REVIEW

File: vuepress/docs/next/docs/reference/deployment-structure.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ The init containers automatically restore a backup included in your application
 The **entando-de-app** is a J2EE application and an instance of the [entando-core](#entando-core). It provides pathways for Entando Components and the server image required by the Entando Operator to manage the deployment. The pom.xml for the application reveals its dependencies.
 
 * GitHub: <https://github.com/entando/entando-de-app/>
-* DockerHub: <https://hub.docker.com/repository/docker/entando/entando-de-app-eap>,<https://hub.docker.com/repository/docker/entando/entando-de-app-wildfly>
+* DockerHub:
```
Comment (jyunmitch) [inline vuepress/docs/next/docs/reference/deployment-structure.md:45]: @pietrangelo need the new DockerHub addr for the 7.2 de-app

File: vuepress/docs/next/tutorials/devops/change-default-datasource.md [hunk 2]
```diff
@@ -1,74 +1,53 @@
...
-If you intend to provide your own Wildfly-based image, please use this 
-[standalone.xml](https://github.com/entando/entando-app-engine-base/blob/develop/wildfly/contrib/wildfly-configuration/standalone.xml)
-file.
+## Customize the Docker Image
+Create a Docker project for your base image using Entando's standard base image. Add the following files to the project folder, typically to a repository like GitHub.  
 
-Please note that if you create your own base image with a different version of EAP or Wildfly, using these files 
-may have unintended consequences.
+**Is server.xml the correct one?**
+1. Using the following Entando template for the Docker image, create the `server.xml`.
```
Comment (jyunmitch) [inline vuepress/docs/next/tutorials/devops/change-default-datasource.md:19]: @pietrangelo I looked inside the folders and made a guess is server.xml the correct base image they should use?

File: vuepress/docs/next/tutorials/devops/change-default-datasource.md [hunk 3]
```diff
@@ -1,74 +1,53 @@
...
+## Customize the Docker Image
+Create a Docker project for your base image using Entando's standard base image. Add the following files to the project folder, typically to a repository like GitHub.  
 
-Please note that if you create your own base image with a different version of EAP or Wildfly, using these files 
-may have unintended consequences.
+**Is server.xml the correct one?**
+1. Using the following Entando template for the Docker image, create the `server.xml`. 
+ 
+If you intend to provide your own Tomcat-based image, please use this 
+[server.xml](https://github.com/entando/entando-app-engine-base/blob/develop/tomcat/conf/server.xml)
```
Comment (jyunmitch) [inline vuepress/docs/next/tutorials/devops/change-default-datasource.md:22]: @pietrangelo should they use this as a base template?

File: vuepress/docs/next/tutorials/devops/change-default-datasource.md [hunk 4]
```diff
@@ -1,74 +1,53 @@
...
+For Tomcat, a basic Dockerfile would look like this:
+**Need new address & names**  
 ```
-FROM entando/entando-eap71-clustered-base:6.1.2
-COPY --chown=185:0 ./standalone-openshift.xml /opt/eap/standalone/configuration
-```  
- 
-For Wildfly, a most basic Dockerfile would look like this:
+FROM entando/entando-XXXX-clustered-base:6.1.2
+COPY --chown=185:0 ./XXX.xml /opt/XXX/standalone/configuration
```
Comment (jyunmitch) [inline vuepress/docs/next/tutorials/devops/change-default-datasource.md:36]: @pietrangelo @nshaw seems like this is a sample we provide for them to look at, do we have a replacement for the tomcat version?

File: vuepress/docs/next/tutorials/devops/change-default-datasource.md [hunk 5]
```diff
@@ -77,77 +56,77 @@ container.
...
+to the previously modified `<image>` element in the Fabric8 Maven Docker Plugin. You can change the `<name>` element
+to reflect your preferred name:
+e.g.,  
+``` bash
+<name>YOUR-DOCKER-REGISTRY.com/YOUR-ORG/YOUR-ENTANDO-APP:1.0.0</name>
+```
+**Does this work for the tomcat version?**
+2. Run:
+``` bash
+mvn clean package -Ptomcat -Pderby
```
Comment (jyunmitch) [inline vuepress/docs/next/tutorials/devops/change-default-datasource.md:102]: @pietrangelo does the run command above work for the  tomcat version?

File: vuepress/docs/next/tutorials/devops/change-default-datasource.md [hunk 6]
```diff
@@ -77,77 +56,77 @@ container.
...
       metadata:
-        name: "my-app"
+        name: "quickstart"
       spec:
         dbms: postgresql
         replicas: 1
-        customServerImage: your-docker-registry.com/your-org/your-entando-app:1.0.0
-        ingressPath: /your-entando-app
+        customServerImage: YOUR-DOCKER-REGISTRY.com/YOUR-ORG/YOUR-ENTANDO-APP:1.0.0
+        ingressPath: /YOUR-ENTANDO-APP
```
Comment (jyunmitch) [inline vuepress/docs/next/tutorials/devops/change-default-datasource.md:129]: @nshaw @pietrangelo for this section, do we check env variables against server.xml or another file? And should we give a list of recommended env variables or just provide sample?

File: vuepress/docs/next/docs/reference/deployment-structure.md [hunk 7]
```diff
@@ -42,7 +42,7 @@ The init containers automatically restore a backup included in your application
 The **entando-de-app** is a J2EE application and an instance of the [entando-core](#entando-core). It provides pathways for Entando Components and the server image required by the Entando Operator to manage the deployment. The pom.xml for the application reveals its dependencies.
 
 * GitHub: <https://github.com/entando/entando-de-app/>
```
Comment (nshaw) [inline vuepress/docs/next/docs/reference/deployment-structure.md:44]: Pretty sure this should be changed to this - https://github.com/entando/app-engine. CC: @pietrangelo @eugeniosant

File: vuepress/docs/next/docs/reference/deployment-structure.md (hunk 1 above)
Comment (nshaw) [reply vuepress/docs/next/docs/reference/deployment-structure.md:45]: https://registry.hub.docker.com/r/entando/entando-de-app-eap, https://registry.hub.docker.com/r/entando/entando-de-app-wildfly, https://registry.hub.docker.com/r/entando/entando-de-app-tomcat

File: vuepress/docs/next/docs/reference/deployment-structure.md [hunk 8]
```diff
@@ -42,7 +42,7 @@ The init containers automatically restore a backup included in your application
...
-* DockerHub: <https://hub.docker.com/repository/docker/entando/entando-de-app-eap>,<https://hub.docker.com/repository/docker/entando/entando-de-app-wildfly>
+* DockerHub: 
 
 #### Customization
 The **entando-de-app** is very likely to be customized as part of an Entando implementation.
```
Comment (nshaw) [inline vuepress/docs/next/docs/reference/deployment-structure.md:48]: We're trying to discourage customization by providing better extension points.
```suggestion
The **entando-de-app** is sometimes customized as part of an Entando implementation. 
```

File: vuepress/docs/next/docs/reference/entandoapp-cr.md [hunk 9]
```diff
@@ -25,19 +25,19 @@ spec:
...
 |`spec.resourceRequirements`| The minimum and maximum [resource allocation](../reference/custom-resources.md#general-resourcerequirements-specifications) for the Entando App Engine container.|
 |`spec.serviceAccountToUse`| The Kubernetes service account in the namespace of the EntandoApp used for the pods hosting the EntandoApps. The default is 'default'.|
-|`spec.standardServerImage`| Either a `tomcat`, `wildfly`, or `eap` image. This property and the `spec.customServerImage` are mutually exclusive. Refer to the [Docker image section](https://github.com/entando-k8s/entando-k8s-controller-coordinator/blob/master/charts/entando-k8s-controller-coordinator/README.md#how-it-resolves-docker-images) to see how the Docker registry and versions are calculated.|
+|`spec.standardServerImage`| `tomcat` is the default image. This property and the `spec.customServerImage` are mutually exclusive. Refer to the [Docker image section](https://github.com/entando-k8s/entando-k8s-controller-coordinator/blob/master/charts/entando-k8s-controller-coordinator/README.md#how-it-resolves-docker-images) to see how the Docker registry and versions are calculated.|
```
Comment (nshaw) [inline vuepress/docs/next/docs/reference/entandoapp-cr.md:40]: `tomcat` -> The `entando-de-app-tomcat` image is the default.

File: vuepress/docs/next/tutorials/devops/build-core-image.md [hunk 10]
```diff
@@ -23,9 +23,9 @@ additional commands and configuration can be found here:
 ## Setup
 
 1.  Clone the application at:
-    <https://github.com/entando/entando-de-app> using
+    <https://github.com/entando/app-engine> using
 
-        git clone https://github.com/entando/entando-de-app
+        git clone https://github.com/entando/app-engine
 
 2.  On a command line, cd into the entando-de-app you just cloned:
```
Comment (nshaw) [inline vuepress/docs/next/tutorials/devops/build-core-image.md:30]: `entando-de-app` -> `app-engine` repository

File: vuepress/docs/next/tutorials/devops/change-default-datasource.md (hunk 4 above)
Comment (nshaw) [reply vuepress/docs/next/tutorials/devops/change-default-datasource.md:36]: The reference version is here - https://github.com/entando/app-engine/blob/develop/Dockerfile.tomcat - and has this line  FROM registry.hub.docker.com/entando/entando-tomcat-base:7.2.0

File: vuepress/docs/next/tutorials/devops/change-default-datasource.md [hunk 11]
```diff
@@ -77,77 +56,77 @@ container.
           </security>
         </datasource>
 ```
-Please take extra care not to remove any lines from these files. This is of particular importance in the case of the 
-EAP image as, on startup, scripts in the official EAP look for certain placeholders in this file and populates them
+**Does this apply to Tomcat**
```
Comment (nshaw) [inline vuepress/docs/next/tutorials/devops/change-default-datasource.md:59]: @pietrangelo thoughts?

File: vuepress/docs/next/tutorials/devops/change-default-datasource.md [hunk 12]
```diff
@@ -77,77 +56,77 @@ container.
...
-### 3. Build your own Docker base image
-You can now build your Docker base image. It is highly recommended that you consider using a dedicated CI/CD build 
-tool such as Jenkins X to build the base image and maintain traceability between your source code and the resulting 
-Docker image. One possible build command could look like this:
-
-`docker build . -t your-docker-registry.com/your-org/your-base-image:1.0.0`
-
-### 4. Modify your Entando App project to use the new Docker base image
+## Build Your Docker Base Image
+It is recommended that you consider using a dedicated CI/CD build tool such as Jenkins X to build the base image and maintain traceability between your source code and the resulting Docker image. One possible build command could look like this:
```
Comment (nshaw) [inline vuepress/docs/next/tutorials/devops/change-default-datasource.md:65]: Jenkins X -> Tekton

File: vuepress/docs/next/tutorials/devops/change-default-datasource.md (hunk 5 above)
Comment (nshaw) [reply vuepress/docs/next/tutorials/devops/change-default-datasource.md:102]: I don't see a tomcat profile - https://github.com/entando/app-engine/blob/develop/webapp/pom.xml. @eugeniosant thoughts?

File: vuepress/docs/next/tutorials/devops/change-default-datasource.md (hunk 2 above)
Comment (nshaw) [reply vuepress/docs/next/tutorials/devops/change-default-datasource.md:19]: @jyunmitch It should be the Docker file - https://github.com/entando/app-engine/blob/release/7.2/Dockerfile.tomcat

File: vuepress/docs/next/tutorials/devops/change-default-datasource.md (hunk 3 above)
Comment (nshaw) [reply vuepress/docs/next/tutorials/devops/change-default-datasource.md:22]: See above

File: vuepress/docs/next/tutorials/devops/change-default-datasource.md (hunk 6 above)
Comment (nshaw) [reply vuepress/docs/next/tutorials/devops/change-default-datasource.md:129]: @pietrangelo One idea @jyunmitch and I had discussed was creating a reference file in the engine repo to contain the list of supported env variables, and then we could point to it from the docs. That would make it easier for the dev team to keep the list in sync with the implementation. CC: @eugeniosant

File: vuepress/docs/next/tutorials/consume/solr.md [hunk 13]
```diff
@@ -75,13 +75,14 @@ kubectl scale deploy/YOUR-APP-NAME-deployment --replicas=0 -n YOUR-NAMESPACE
 
 2. Edit the deployment and add the following environment variables:
 
-``` bash
+``` yaml
 spec
```
Comment (nshaw) [inline vuepress/docs/next/tutorials/consume/solr.md:79]: Missing a colon here too

File: vuepress/docs/next/tutorials/consume/solr.md [hunk 14]
```diff
@@ -75,13 +75,14 @@ kubectl scale deploy/YOUR-APP-NAME-deployment --replicas=0 -n YOUR-NAMESPACE
...
-     - name: SOLR_ACTIVE
-       value: "true"
-     - name: SOLR_ADDRESS
-	     value: http://solr-solrcloud-common/solr 
+   containers:
+     - env
+       - name: SOLR_ACTIVE
+         value: "true"
+       - name: SOLR_ADDRESS
+	       value: http://solr-solrcloud-common/solr
```
Comment (nshaw) [inline vuepress/docs/next/tutorials/consume/solr.md:85]: Indentation looks off here, based on the git diff there's some odd characters. The name and value lines should start at the same column,

Comment (jyunmitch) [reply vuepress/docs/next/tutorials/consume/solr.md:85]: oddly, it looks fine in vs code.

**Review by nshaw — APPROVED** [review]

**Pull request merged**